src/glue/jobs/json_to_parquet.py

```diff
@@ -0,0 +1,319 @@
+assessmentid, year, month, and day to each record in each table.
+"""
+
+import os
+import sys
+import boto3
+
+from awsglue import DynamicFrame
+from awsglue.context import GlueContext
+from awsglue.job import Job
+from awsglue.gluetypes import StructType
+from awsglue.utils import getResolvedOptions
+from pyspark import SparkContext
+
+INDEX_FIELD_MAP = {
+    "enrolledparticipants": ["ParticipantIdentifier"],
+    "fitbitprofiles": ["ParticipantIdentifier", "ModifiedDate"],
+    "fitbitdevices": ["ParticipantIdentifier", "Date"],
+    "fitbitactivitylogs": ["LogId"],
+    "fitbitdailydata": ["ParticipantIdentifier", "Date"],
+    "fitbitintradaycombined": ["ParticipantIdentifier", "Type", "DateTime"],
+    "fitbitrestingheartrates": ["ParticipantIdentifier", "Date"],
+    "fitbitsleeplogs": ["LogId"],
+    "healthkitv2characteristics": ["HealthKitCharacteristicKey"],
+    "healthkitv2samples": ["HealthKitSampleKey"],
+    "healthkitv2samples_deleted": ["HealthKitSampleKey"],
+    "healthkitv2heartbeat": ["HealthKitHeartbeatSampleKey"],
+    "healthkitv2statistics": ["HealthKitStatisticKey"],
+    "healthkitv2clinicalrecords": ["HealthKitClinicalRecordKey"],
+    "healthkitv2electrocardiogram": ["HealthKitECGSampleKey"],
+    "healthkitv2workouts": ["HealthKitWorkoutKey"],
+    "healthkitv2activitysummaries": ["HealthKitActivitySummaryKey"],
+    "googlefitsamples": ["GoogleFitSampleKey"],
+    "symptomlog": ["DataPointKey"]
+}
+
+def get_args() -> tuple[dict, dict]:
+    """
+    Get job and workflow arguments.
+
+    Returns:
+        (tuple) : (job arguments (dict), workflow run properties (dict))
+    """
+    glue_client = boto3.client("glue")
+    args = getResolvedOptions(
+             sys.argv,
+             ["WORKFLOW_NAME",
+              "WORKFLOW_RUN_ID",
+              "JOB_NAME",
+              "glue-table"])
+    workflow_run_properties = glue_client.get_workflow_run_properties(
+            Name=args["WORKFLOW_NAME"],
+            RunId=args["WORKFLOW_RUN_ID"])["RunProperties"]
+    return args, workflow_run_properties
+
+def has_nested_fields(schema: StructType) -> bool:
+    """
+    Determine whether a DynamicFrame schema has struct or array fields.
+
+    If the DynamicFrame does not have fields of these data types, it is flat and
+    can be written directly to S3.  If it does, then the DynamicFrame will need
+    to be 'relationalized' so that all the data contained in the DynamicFrame has
+    been flattened.
+
+    Args:
+        schema (awsglue.gluetypes.StructType): The schema of a DynamicFrame.
+
+    Returns:
+        bool: Whether this schema contains struct or array fields.
+    """
+    for col in schema:
+        if col.dataType.typeName() in ["array", "struct"]:
+            return True
+    return False
+
+def get_table(
+        table_name: str,
+        database_name: str,
+        glue_context: GlueContext
+    ) -> DynamicFrame:
+    """
+    Return a table as a DynamicFrame with an unambiguous schema.
```

Comment:
What do you mean by this? How does dropping the duplicates make the schema umambiguous?

Reply:
It doesn't. Dynamic Frames have an "ambiguous" schema in the sense that they do not enforce a schema on the individual Dynamic Records. To fix the schema (more specifically, to fix the data types), you need to call `resolveChoice` on the Dynamic Frame.